src/algorithms/reco/JetReconstruction.cc

```diff
@@ -55,6 +56,19 @@ namespace eicrecon {
     // Choose jet definition based on no. of parameters
     switch (m_mapJetAlgo[m_cfg.jetAlgo]) {
 
+      // contributed algorithms
+      case JetAlgorithm::plugin_algorithm:
+
+        // expand to other algorithms as required
+        if(m_cfg.jetContribAlgo == "Centauro"){
+          m_jet_def = std::make_unique<JetDefinition>(new contrib::CentauroPlugin(m_cfg.rJet));
```

Comment:
Somehow this is detected as a memory leak:
```
Direct leak of 64 byte(s) in 2 object(s) allocated from:
    #0 0x55f986acb1fd in operator new(unsigned long) (/home/runner/work/EICrecon/EICrecon/install/bin/eicrecon+0x10e1fd) (BuildId: b5a759ebf1114ef904000411f83b34f5cafce8b5)
    #1 0x7f5868a08f7c in eicrecon::JetReconstruction<edm4eic::ReconstructedParticle>::init(std::shared_ptr<spdlog::logger>) /home/runner/work/EICrecon/EICrecon/src/algorithms/reco/JetReconstruction.cc:64:55
    #2 0x7f58691db696 in eicrecon::JetReconstruction_factory<edm4eic::ReconstructedParticle>::Configure() /home/runner/work/EICrecon/EICrecon/src/factories/reco/JetReconstruction_factory.h:49:17
    #3 0x7f58703cdf96 in __pthread_once_slow nptl/pthread_once.c:116:7
```

Reply:
I don't see anything suspicious in the `include/fastjet/contrib/Centauro.hh`:
```c++
    CentauroPlugin (double R) : _R(R), _gammaE(0), _gammaPz(0){}
```

Reply:
I see that `JetDefinition::Plugin` has a virtual destructor and none in `class CentauroPlugin : public JetDefinition::Plugin {`. I wonder, if this can be the cause?

Reply:
Can you store the plugin in `std::unique_ptr m_jet_plugin` so its ownership is clear?

Reply:
Right. The JetDefinition is constructed from it, but we don't store the pointer to the plugin.
```c++
  /// constructor based on a pointer to a user's plugin; the object
  /// pointed to must remain valid for the whole duration of existence
  /// of the JetDefinition and any related ClusterSequences
  JetDefinition(const Plugin * plugin_in) {
```

Reply:
Thanks guys - I had a similar problem trying to do this before, it yields error messages I don't understand.  

In JetReconstruction.h I declared: 

`      
std::unique_ptr<fastjet::JetDefinition::Plugin> m_jet_plugin;
`
and then in the code I use this as: 

`
        if(m_cfg.jetContribAlgo == "Centauro"){
	  m_jet_plugin = std::make_unique<contrib::CentauroPlugin>(m_cfg.rJet); 
          m_jet_def = std::make_unique<JetDefinition>(m_jet_plugin);
        }
`

This yields the error at compilation: 

`
/home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.cc: In member function ‘void eicrecon::JetReconstruction<InputT>::init(std::shared_ptr<spdlog::logger>)’:
/home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.cc:64:26: error: expected primary-expression before ‘const’
   64 |           m_jet_plugin = const std::make_unique<contrib::CentauroPlugin>(m_cfg.rJet);
      |                          ^~~~~
In file included from /usr/include/c++/12/memory:76,
                 from /usr/local/include/fmt/format.h:40,
                 from /usr/local/include/algorithms/detail/demangle.h:9,
                 from /usr/local/include/algorithms/algorithm.h:39,
                 from /home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.h:6,
                 from /home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.cc:5:
/usr/include/c++/12/bits/unique_ptr.h: In instantiation of ‘std::__detail::__unique_ptr_t<_Tp> std::make_unique(_Args&& ...) [with _Tp = fastjet::JetDefinition; _Args = {const unique_ptr<fastjet::JetDefinition::Plugin, default_delete<fastjet::JetDefinition::Plugin> >&}; __detail::__unique_ptr_t<_Tp> = __detail::__unique_ptr_t<fastjet::JetDefinition>]’:
/home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.cc:65:54:   required from ‘void eicrecon::JetReconstruction<InputT>::init(std::shared_ptr<spdlog::logger>) [with InputT = edm4eic::ReconstructedParticle]’
/home/lajoie/ePIC/EICrecon/src/algorithms/reco/JetReconstruction.cc:160:18:   required from here
/usr/include/c++/12/bits/unique_ptr.h:1065:30: error: no matching function for call to ‘fastjet::JetDefinition::JetDefinition(const std::unique_ptr<fastjet::JetDefinition::Plugin>&)’
 1065 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
`
which I think is telling me I'm using the pointer wrong? 

Any pointers? (pun intended)

Reply:
Try with
```
m_jet_def = std::make_unique<JetDefinition>(m_jet_plugin.get())
```

Reply:
The `get()` gives you a regular pointer from the smart pointer.

Reply:
Well that seems to be the secret sauce!